src/IO/H5/Helpers.cpp

```diff
@@ -20,6 +21,79 @@
 #include "Utilities/Gsl.hpp"
 #include "Utilities/MakeArray.hpp"
 
+namespace {
+// Given either a std::vector or DataVector, converts data to a
+// scalar (double or int), std::vector, DataVector, or boost::multiarray
+template<size_t Rank, typename T>
+struct VectorToArray {};
+
+template<typename T>
+struct VectorToArray<0,T> {
+  T vector_to_array(const std::vector<T>& raw_data,
+    const std::array<hsize_t, 1>& /*size*/) noexcept {
+    return raw_data[0];
+ }
+};
+
+template<typename T>
+struct VectorToArray<1,T> {
+  T vector_to_array(const T& raw_data,
+    const std::array<hsize_t, 1>& /*size*/) noexcept {
+    return raw_data;
```

Comment:
@wthrowe I think for function arguments you want to `std::move` in the return because copy-elision doesn't work, right?

Reply:
I thought so, but I did a quick test with clang (it gives warnings if you obviously screw these up) and it reports a move of a local as a _pessimizing move_ (prevents copy elision) and a move of an argument as a _redundant move_ (harmless but would have been done anyway).

The place we encountered this before was in an old version of `lift_flux`.  There were two things going on there, which is probably why we are misremembering.  There was returning the parameter, which prevented copy elision but not moving, and returning a reference variable, which causes a copy be default (and so requires a `std::move`).

Reply:
Ah okay, that makes sense. Thanks for checking and clarifying! :)